src/resolvers/supersource.ts

```diff
@@ -1,14 +1,33 @@
 import {
-	Commands as AtemCommands, VideoState
+	Commands as AtemCommands, VideoState, Enums
 } from 'atem-connection'
 import { State as StateObject } from '..'
 
-export function resolveSupersourceBoxState (oldState: StateObject, newState: StateObject): Array<AtemCommands.AbstractCommand> {
+export function resolveSuperSourceState (oldState: StateObject, newState: StateObject, version: Enums.ProtocolVersion): Array<AtemCommands.AbstractCommand> {
+	let commands: Array<AtemCommands.AbstractCommand> = []
+
+	if (!newState.video.superSources) return commands
+
+	if (version < Enums.ProtocolVersion.V8_0) {
+		commands = commands.concat(
+			resolveSuperSourceBoxState(oldState, newState),
+			resolveSuperSourcePropertiesState(oldState, newState))
+	} else {
+		commands = commands.concat(
+			resolveSuperSourceBoxState(oldState, newState),
+			resolveSuperSourcePropertiesV8State(oldState, newState),
+			resolveSuperSourceBorderV8State(oldState, newState))
+	}
+
+	return commands
+}
+
+export function resolveSuperSourceBoxState (oldState: StateObject, newState: StateObject): Array<AtemCommands.AbstractCommand> {
 	const commands: Array<AtemCommands.AbstractCommand> = []
 
-	for (const index in newState.video.superSourceBoxes) {
-		const newBox = newState.video.superSourceBoxes[index] || {}
-		const oldBox = oldState.video.superSourceBoxes[index] || {}
+	for (const index in newState.video.superSources[0].boxes) {
```

Comment:
`resolveSuperSourceBoxState` is used for v8, so needs to iterate through all the supersources